### PR TITLE
Removed the specified colors in graphs

### PR DESCRIPTION
**Pull request by adutfoy**
This PR tackles the issue #2543 .

- I removed the color specifications when the default color were useds.
- I also modified LagrangeP1Evaluation: the pyplot was not displayed: https://openturns.github.io/openturns/latest/user_manual/_generated/openturns.P1LagrangeEvaluation.html?highlight=p1lagrangeevaluation

### CODE REVIEW

File: python/doc/examples/meta_modeling/fields_metamodels/plot_viscous_fall_metamodel.py [hunk 1]
```diff
@@ -146,4 +146,4 @@
 ot.ResourceMap.Reload()
 
 # %%
-otv.View.ShowAll()
+ot.View.ShowAll()
```
Comment (josephmure) [inline python/doc/examples/meta_modeling/fields_metamodels/plot_viscous_fall_metamodel.py:149]:
```suggestion
otv.View.ShowAll()
```
the original line was correct, View belongs to the openturns.viewer module, not to openturns directly;

File: python/doc/examples/meta_modeling/general_purpose_metamodels/plot_general_linear_model.py [hunk 2]
```diff
@@ -63,14 +63,14 @@
 xmax = 5.0
 x1value = 0.5
 parametricModelGraph = ot.ParametricFunction(model, [0], [x1value]).draw(xmin, xmax)
-parametricModelGraph.setColors(["red"])
 graphMetamodel = ot.ParametricFunction(responseSurface, [0], [x1value]).draw(xmin, xmax)
-graphMetamodel.setColors(["blue"])
```
Comment (josephmure) [inline python/doc/examples/meta_modeling/general_purpose_metamodels/plot_general_linear_model.py:68]: Unsure about this deletion, were the colors not specified in order to override the default in this particular case ?

File: python/doc/examples/data_analysis/distribution_fitting/plot_estimate_conditional_quantile.py [hunk 3]
```diff
@@ -107,7 +107,7 @@
 graph.add(cxy_ex)
 graph.setLegends(["$Q_2$ kernel smoothing", "$Q_2$ exact"])
 graph.setLegendPosition("lower right")
-graph.setColors(["red", "blue"])
+#graph.setColors(["red", "blue"])
```
Comment (josephmure) [inline python/doc/examples/data_analysis/distribution_fitting/plot_estimate_conditional_quantile.py:110]: Unsure about this deletion, were the colors not specified in order to override the default in this particular case ?

File: python/doc/examples/data_analysis/distribution_fitting/plot_estimate_conditional_quantile.py [hunk 4]
```diff
@@ -193,10 +193,12 @@
...
```
Comment (josephmure) [inline python/doc/examples/data_analysis/distribution_fitting/plot_estimate_conditional_quantile.py:196]: Unsure about this deletion, were the colors not specified in order to override the default in this particular case ?

File: python/doc/examples/functional_modeling/field_functions/plot_process_sample.py [hunk 5]
```diff
@@ -149,7 +149,6 @@
     graph.add(curve)
 
 curve = ot.Curve(vertices, process_sample_mean, "process sample mean")
-curve.setColor("red")
```
Comment (josephmure) [inline python/doc/examples/functional_modeling/field_functions/plot_process_sample.py:152]: in this particular case, there are already n_samples curves on the graph, so removing this line will make the curve use an arbitrary color

File: python/doc/pyplots/Dirichlet.py [hunk 6]
```diff
@@ -10,16 +10,14 @@
...
 list_theta = [(2, 2), (7, 5), (2, 6), (3, 4)]
 for theta in list_theta:
     distribution = ot.Dirichlet(theta)
     pdf_curve = distribution.drawPDF()
     cdf_curve = distribution.drawCDF()
     pdf_graph.add(pdf_curve)
     cdf_graph.add(cdf_curve)
-pdf_graph.setColors([palette[i] for i in range(4)])
-cdf_graph.setColors([palette[i] for i in range(4)])
+cdf_graph.setColors(pdf_graph.getColors())
```
Comment (josephmure) [inline python/doc/pyplots/Dirichlet.py:20]: even this line might no longer be necessary if pdf_graph and cdf_graph both use the default palette

File: python/doc/pyplots/GeneralizedExtremeValue.py [hunk 7]
```diff
@@ -15,7 +15,6 @@
     pdf_curve.setLegend(legend)
     cdf_curve.setLegend(legend)
     pdf_graph.add(pdf_curve)
-    cdf_graph.add(cdf_curve)
```
Comment (josephmure) [inline python/doc/pyplots/GeneralizedExtremeValue.py:18]: why this deletion ?

File: python/doc/pyplots/P1LagrangeEvaluation.py [hunk 8]
```diff
@@ -12,9 +12,9 @@
 cloud = ot.Cloud(mesh.getVertices(), values)
 cloud.setPointStyle("square")
 graph.add(cloud)
-graph.setColors(["blue", "red"])
 graph.setTitle("P1 Lagrange interpolation")
 
 fig = plt.figure(figsize=(10, 4))
 func_axis = fig.add_subplot(111)
-view = View(graph, figure=fig, axes=[func_axis], add_legend=False)
+View(graph, figure=fig, axes=[func_axis], add_legend=False)
```
Comment (josephmure) [inline python/doc/pyplots/P1LagrangeEvaluation.py:19]: I think this is likely to result in an unsightly print on the HTML page, but I may be wrong

File: python/src/TrendFactory_doc.i.in [hunk 9]
```diff
@@ -60,7 +60,7 @@ Examples
 Define a scalar temporal Gaussian process on a mesh of dimension 1:
 
 >>> import openturns as ot
->>> myGrid = ot.RegularGrid(0.0, 1.0, 100)
+>>> myGrid = ot.RegularGrid(0.0, 0.1, 100)
```
Comment (josephmure) [inline python/src/TrendFactory_doc.i.in:63]: why this change ?

**Review by josephmure — CHANGES_REQUESTED** [review]
Thanks for cleaning up the doc! I only have a few concerns.

File: python/doc/examples/meta_modeling/fields_metamodels/plot_viscous_fall_metamodel.py (hunk 1 above)
Comment (adutfoy) [reply python/doc/examples/meta_modeling/fields_metamodels/plot_viscous_fall_metamodel.py:149]: yes

File: python/doc/pyplots/Dirichlet.py (hunk 6 above)
Comment (adutfoy) [reply python/doc/pyplots/Dirichlet.py:20]: yes!

File: python/doc/pyplots/P1LagrangeEvaluation.py (hunk 8 above)
Comment (adutfoy) [reply python/doc/pyplots/P1LagrangeEvaluation.py:19]: I will check.

File: python/src/TrendFactory_doc.i.in (hunk 9 above)
Comment (adutfoy) [reply python/src/TrendFactory_doc.i.in:63]: You are right: No importance actually. I just wanted to make the grid coherent with the next graph which contains the Trend drawn in [0, 10]. 
I removed my modification.

File: python/doc/pyplots/GeneralizedExtremeValue.py (hunk 7 above)
Comment (adutfoy) [reply python/doc/pyplots/GeneralizedExtremeValue.py:18]: an error. I removed also the lines that were specifying the color of each Drawable.

File: python/doc/examples/functional_modeling/field_functions/plot_process_sample.py (hunk 5 above)
Comment (adutfoy) [reply python/doc/examples/functional_modeling/field_functions/plot_process_sample.py:152]: yes! But I saw that all the process samples are idential! It come sfrom the fact that the X_distribution is a JointDistribution composed of marginals which are Dirac distributions...:
```
>>>print(X_distribution)
JointDistribution(Dirac(point = [7.5e+08]), Dirac(point = [2.75e+09]), Dirac(point = [10]), IndependentCopula(dimension = 3))
```
 These marginals come from the Caboche model.

Comment (adutfoy) [reply python/doc/examples/functional_modeling/field_functions/plot_process_sample.py:152]: yes! But I saw that all the process samples are idential! It come sfrom the fact that the X_distribution is a JointDistribution composed of marginals which are Dirac distributions...:
```
>>>print(X_distribution)
JointDistribution(Dirac(point = [7.5e+08]), Dirac(point = [2.75e+09]), Dirac(point = [10]), IndependentCopula(dimension = 3))
```
 These marginals come from the Caboche model.

Comment (adutfoy) [reply python/doc/examples/functional_modeling/field_functions/plot_process_sample.py:152]: This example is not mentioned in the section 'Examples based on this usecase' in the Chaboche usecase! Actually, I did not manage to find it in the OT documentation....

**Comment by adutfoy** [comment]
@jschueller : I don't know how to simplify that PR to keep only what I modified....

**Comment by adutfoy** [comment]
I don't understand what the compilation problem is about... I need help! @jschueller

**Comment by jschueller** [comment]
there are now "import matplotlib..." lines in the examples you modified that are useless since matplotlib is not directly used anymore, just remove the designated lines and it will be fine

File: python/doc/pyplots/P1LagrangeEvaluation.py [hunk 10]
```diff
@@ -12,9 +12,8 @@
...
```
Comment (m-balesdent) [inline python/doc/pyplots/P1LagrangeEvaluation.py:19]: View(graph, figure=fig, axes=[func_axis], add_legend=False).show()

I think show() is necessary

**Review by m-balesdent — CHANGES_REQUESTED** [review]

File: python/doc/examples/functional_modeling/field_functions/plot_process_sample.py (hunk 5 above)
Comment (josephmure) [reply python/doc/examples/functional_modeling/field_functions/plot_process_sample.py]: I don't know if you did anything, but it is now mentioned in the "Examples based on this usecase" section: see https://output.circle-artifacts.com/output/job/ad1e2baf-1c02-4e4a-9e1f-4703618ecc26/artifacts/0/html/usecases/use_case_chaboche.html#examples-based-on-this-use-case

File: python/doc/pyplots/P1LagrangeEvaluation.py (hunk 8 above)
Comment (josephmure) [reply python/doc/pyplots/P1LagrangeEvaluation.py:19]: I was wrong, because this is an API page illustration script, not an example script.

File: python/doc/pyplots/P1LagrangeEvaluation.py (hunk 10 above)
Comment (josephmure) [reply python/doc/pyplots/P1LagrangeEvaluation.py:19]: No it is not, this is designed to produce the plot on the API doc page. The reason it does not is that this page uses the "class" template instead of the "classWithPlot" template. @adutfoy to fix that please insert 
`:template: classWithPlot.rst_t`
above "P1LagrangeEvaluation" in `python/doc/user_manual/functions.rst` :

image

Comment (m-balesdent) [reply python/doc/pyplots/P1LagrangeEvaluation.py:19]: Maybe there are other classes that do not use "classWithPlot" in the library?

File: python/doc/examples/functional_modeling/field_functions/plot_process_sample.py (hunk 5 above)
Comment (adutfoy) [reply python/doc/examples/functional_modeling/field_functions/plot_process_sample.py]: ok! It appears now!

File: python/doc/user_manual/functions.rst [hunk 11]
```diff
@@ -83,6 +83,8 @@ Refer to :ref:`field_function`.
     OpenTURNSPythonFieldFunction
     OpenTURNSPythonFieldToPointFunction
     OpenTURNSPythonPointToFieldFunction
+
+    :template: classWithPlot.rst_t
     P1LagrangeInterpolation
```
Comment (jschueller) [inline python/doc/user_manual/functions.rst:88]: if you want to use classWithPlot template then you should provide the python/doc/pyplots/P1LagrangeInterpolation.py which is missing, maybe you forgot to "git add" it ?

Comment (adutfoy) [reply python/doc/user_manual/functions.rst:88]: There was 2 errors here!

1. the P1LagrangeInterpolation does not exist : I removed the line from the function.rst
2. the P1LagrangeEvaluation was not associated to a classWithPlot: I introduced the line

**Review by m-balesdent — APPROVED** [review]
Thank you for the job !

File: python/doc/user_manual/functions.rst [hunk 12]
```diff
@@ -83,7 +83,6 @@ Refer to :ref:`field_function`.
     OpenTURNSPythonFieldFunction
     OpenTURNSPythonFieldToPointFunction
     OpenTURNSPythonPointToFieldFunction
-    P1LagrangeInterpolation
```
Comment (jschueller) [inline python/doc/user_manual/functions.rst:86]: it must not be removed, but you can move it below near P1LagrangeEvaluation

File: python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py [hunk 13]
```diff
@@ -604,6 +604,3 @@ def func(X):
 grid.setGraph(1, 1, graph)
 
 _ = View(grid)
-
-
-View.ShowAll()
```
Comment (jschueller) [inline python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py:609]: this should not be removed (restore it in a new cell with %%)

File: python/doc/examples/calibration/bayesian_calibration/plot_rwmh_python_distribution.py [hunk 14]
```diff
@@ -244,5 +244,3 @@ def getParameter(self):
     graph.setLegends(["Posterior"])
     grid.setGraph(0, parameter_index, graph)
 _ = View(grid)
-
-View.ShowAll()
```
Comment (jschueller) [inline python/doc/examples/calibration/bayesian_calibration/plot_rwmh_python_distribution.py:248]: this one too

**Comment by jschueller** [comment]
just two small issues and it can be merged

File: python/doc/user_manual/functions.rst (hunk 12 above)
Comment (adutfoy) [reply python/doc/user_manual/functions.rst:86]: I let it where it was...

File: python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py (hunk 13 above)
Comment (adutfoy) [reply python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py:609]: sorry... It is done!
